### ISSUE TITLE
can not receive call when CALL WAITING is disabled

### ISSUE THREAD

**Issue by edimondi** [InnovateAsterisk/Browser-Phone#178]
hi there, i updated the browser phone to latest version and i am having some issue for example if i disable CALL WAITING than i don't receive calls than it works for a while than it stoped working i dont know where is the issue.. here is the message log:

**Show Popup Menu
phone.js:10567 CallWaiting false
sip-0.20.0.min.js:2 Thu Dec 02 2021 12:35:33 GMT+0000 (Coordinated Universal Time) | sip.Invitation | Invitation.progress
sip-0.20.0.min.js:2 Thu Dec 02 2021 12:35:33 GMT+0000 (Coordinated Universal Time) | sip.invite-dialog | INVITE dialog 07de75dd-3b96-42c5-8b1a-d90a02e140da1op7g5v90i883ac4c9-79e2-47a6-af6c-74a0d7925315 constructed
phone.js:1415 New Incoming Call! Demo <340>
phone.js:1418 Current Call Count: 2
phone.js:1512 Call Waiting Disabled, rejecting call.
sip-0.20.0.min.js:2 Thu Dec 02 2021 12:35:33 GMT+0000 (Coordinated Universal Time) | sip.Invitation | Invitation.reject
sip-0.20.0.min.js:2 Thu Dec 02 2021 12:35:33 GMT+0000 (Coordinated Universal Time) | sip.Invitation | Session 07de75dd-3b96-42c5-8b1a-d90a02e140da883ac4c9-79e2-47a6-af6c-74a0d7925315 transitioned to state Terminated
sip-0.20.0.min.js:2 Thu Dec 02 2021 12:35:33 GMT+0000 (Coordinated Universal Time) | sip.Invitation | Session 07de75dd-3b96-42c5-8b1a-d90a02e140da883ac4c9-79e2-47a6-af6c-74a0d7925315 in state Terminated is being disposed
phone.js:4091 New CDR {CdrId: '16384485335001E5', ItemType: 'CDR', ItemDate: '2021-12-02 12:35:33 UTC', CallAnswer: null, CallEnd: '2021-12-02 12:35:33 UTC', …}
phone.js:4282 Last Activity for Demo is now: 2021-12-02 12:35:33 UTC
sip-0.20.0.min.js:2 Thu Dec 02 2021 12:35:33 GMT+0000 (Coordinated Universal Time) | sip.invite-dialog | INVITE dialog 07de75dd-3b96-42c5-8b1a-d90a02e140da1op7g5v90i883ac4c9-79e2-47a6-af6c-74a0d7925315 destroyed**

**Comment by InnovateAsterisk** [comment]
Hmmm I see what the problem is... I was worried about this. In the new version they dropped support for the userAgent.session array - that keeps an up-to-date array of active/valid sessions. I used a bit of a hack, but I think I may need to re-do that part. 

The main problem comes in with the onInviteCancel for an inbound call - their session waits for an an ACK from the OK when a call is CANCELed, but Asterisk doesn't give it or their don't pick it up.... So it times-out, complains, and then you should see something like:

``` 
 ... | sip.transaction.ist | ACK to negative final response was never received, terminating transaction....
 ... | sip.invite-dialog | INVITE dialog b4987e5-613319b14ad9 destroyed
```

And only after the INVITE is destroyed, so the array updated... so the browser phone thinks you have an active call.

Notice in the line above, you have ```phone.js:1418 Current Call Count: 2``` but you will probably find that was not really the case.... it was probably waiting for that final ACK.

**Comment by edimondi** [comment]
the changes that you made yesterday broke the browser phone completely, now i see blank page

**Comment by InnovateAsterisk** [comment]
Does the console show an error?

**Comment by edimondi** [comment]
jQuery.Deferred exception: phoneOptions is not defined ReferenceError: phoneOptions is not defined
    at HTMLDocument.<anonymous> (https://cmydomain/phone.js:336:19)
    at l (https://mydomain/lib/jquery/jquery-3.3.1.min.js:2:29375)
    at c (https://cmydomain/lib/jquery/jquery-3.3.1.min.js:2:29677) undefined
w.Deferred.exceptionHook @ jquery-3.3.1.min.js:2
jquery-3.3.1.min.js:2 Uncaught ReferenceError: phoneOptions is not defined
    at HTMLDocument.<anonymous> (phone.js:336)
    at l (jquery-3.3.1.min.js:2)
    at c (jquery-3.3.1.min.js:2)

**Comment by InnovateAsterisk** [comment]
Ah, ok, that I need to look at. For the moment, take the index.html file too. If you have made your own index page, make sure you have defined a global variable: phoneOptions.

You can now provision the options from the index.html

**Comment by InnovateAsterisk** [comment]
> jQuery.Deferred exception: phoneOptions is not defined ReferenceError: phoneOptions is not defined
at HTMLDocument. (https://cmydomain/phone.js:336:19)
at l (https://mydomain/lib/jquery/jquery-3.3.1.min.js:2:29375)
at c (https://cmydomain/lib/jquery/jquery-3.3.1.min.js:2:29677) undefined
w.Deferred.exceptionHook @ jquery-3.3.1.min.js:2
jquery-3.3.1.min.js:2 Uncaught ReferenceError: phoneOptions is not defined
at HTMLDocument. (phone.js:336)
at l (jquery-3.3.1.min.js:2)
at c (jquery-3.3.1.min.js:2)

This issue is now fixed, see: 
https://github.com/InnovateAsterisk/Browser-Phone/commit/fb8643cfa0566635d0176fc73d2fa76e1d6a017a
